### PR TITLE
Keycloak refresh done using tokenservices

### PR DESCRIPTION
**Pull request by saravankumarpa**
Token service timer is initiated on app load and sign in

*Issue #:*
https://app.zenhub.com/workspaces/entity-5bf2f2164b5806bc2bf60531/issues/bcgov/entity/1773

*Description of changes:*
The token service is invoked on app load and sign in as well.  Interceptor code also checked in ;but not used right now because of  https://github.com/bcgov/entity/issues/2076
Need more testing in DEV .
*Checklist:*
I confirm that below checklist items are addressed in this pull request; (check the boxes applicable)
- [ ] No lint errors
- [ ] No test case failures and proper coverage for all modules/classes
- [ ] Updated the deployment configs for new environment variable(s)
- [ ] Updtaed the postman collection in entity repository (https://github.com/bcgov/entity/tree/master/api-e2e/postman) for e2e tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/components/auth/Signin.vue [hunk 1]
```diff
@@ -52,6 +52,12 @@ export default class Signin extends Mixins(NextPageMixin) {
         if (this.idpHint === 'bcsc') {
           await this.syncUserProfile()
           await this.syncOrganizations()
+          // eslint-disable-next-line no-console
+          console.info('[SignIn.vue]Logged in User.Starting refreshTimer')
+          var self = this
+          this.$tokenService.initUsingUrl(`/${process.env.VUE_APP_PATH}/config/kc/keycloak.json`).then(function (success) {
```
Comment (sutherlanda) [inline auth-web/src/components/auth/Signin.vue]: My preference is using `async/await` instead for asynchronous.  Do you have any objections if we stick to that pattern?

File: auth-web/src/services/keycloak.services.ts [hunk 2]
```diff
@@ -78,6 +79,10 @@ class KeyCloakService {
     }
   }
 
+  getKCInstance () :KeycloakInstance {
```
Comment (sutherlanda) [inline auth-web/src/services/keycloak.services.ts]: Spacing

`getKCInstance (): KeycloakInstance`

File: auth-web/src/util/constants.ts [hunk 3]
```diff
@@ -2,6 +2,7 @@
 export enum SessionStorageKeys {
     KeyCloakToken = 'KEYCLOAK_TOKEN',
     KeyCloakRefreshToken = 'KEYCLOAK_REFRESH_TOKEN',
+    Idtoken = 'KEYCLOAK_ID_TOKEN',
```
Comment (sutherlanda) [inline auth-web/src/util/constants.ts]: `IdToken` (sorry I'm being picky)

File: auth-web/src/util/interceptors.ts [hunk 4]
```diff
@@ -14,4 +15,33 @@ export default function setup () {
   }, function (err) {
     return Promise.reject(err)
   })
+  // enable to do an interceptor which will refresh the token on 401 failures
+  // createAxiosResponseInterceptor()
```
Comment (sutherlanda) [inline auth-web/src/util/interceptors.ts]: Leftover comment

File: auth-web/src/util/interceptors.ts [hunk 5]
```diff
@@ -14,4 +15,33 @@ export default function setup () {
...
+      const originalRequest = error.config
+      if (error.response.status !== 401) {
+        return new Promise((resolve, reject) => {
+          reject(error)
+        })
+      }
+      axios.interceptors.response.eject(interceptor)
+      if (error.response.status === 401 && !originalRequest._retry) {
+        originalRequest._retry = true
+        let mm :TokenService = new TokenService()
```
Comment (sutherlanda) [inline auth-web/src/util/interceptors.ts]: Explicit typing not necessary here.

File: auth-web/src/components/auth/Signin.vue (hunk 1 above)
Comment (saravankumarpa) [reply auth-web/src/components/auth/Signin.vue]: Will change it

File: auth-web/src/App.vue [hunk 6]
```diff
@@ -23,6 +23,18 @@ export default Vue.extend({
     SbcHeader,
     SbcFooter,
     PaySystemAlert
+  },
+  mounted (): void {
+    // eslint-disable-next-line no-console
+    console.log('APP.vue mounted')
```
[the remark this answers is not in the document]
Comment (saravankumarpa) [reply auth-web/src/App.vue]: Just till we get this dev :-) , after that i will remove

File: auth-web/src/components/auth/Signin.vue (hunk 1 above)
Comment (saravankumarpa) [reply auth-web/src/components/auth/Signin.vue]: just the habit :-( .. changing it

File: auth-web/src/App.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (sutherlanda) [reply auth-web/src/App.vue]: No problem

**Pull request merged**